src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2820,31 +2820,42 @@ private TypeWithState VisitCallReceiver(BoundCall node)
             return receiverType;
         }
 
-        private VisitResult GetReturnType(MethodSymbol method)
+        private TypeWithState GetReturnTypeWithState(MethodSymbol method)
         {
-            var type = method.ReturnTypeWithAnnotations;
-            var rvalueType = TypeWithState.Create(type, GetAnnotations(method));
-            return new VisitResult(rvalueType, type);
+            return TypeWithState.Create(method.ReturnTypeWithAnnotations, GetRValueAnnotations(method));
         }
 
-        private FlowAnalysisAnnotations GetAnnotations(ParameterSymbol parameter)
+        private FlowAnalysisAnnotations GetRValueAnnotations(Symbol symbol)
         {
             // Annotations are ignored when binding an attribute to avoid cycles. (Members used
             // in attributes are error scenarios, so missing warnings should not be important.)
-            return IsAnalyzingAttribute ?
-                FlowAnalysisAnnotations.None :
-                parameter.FlowAnalysisAnnotations;
+            if (IsAnalyzingAttribute)
+            {
+                return FlowAnalysisAnnotations.None;
+            }
+
+            var annotations = symbol switch
+            {
+                MethodSymbol method => method.ReturnTypeFlowAnalysisAnnotations,
+                PropertySymbol property => property.GetMethod?.ReturnTypeFlowAnalysisAnnotations ?? FlowAnalysisAnnotations.None,
```

Comment:
>property.GetMethod?. [](start = 43, length = 20)

It doesn't look like this code will properly handle overridden properties, when overriding property overrides only setter.